simulation/optimization/genetic.py

```diff
@@ -0,0 +1,611 @@
+                 elitism: int = 8,
+                 mutation_type: Mutation_Type = Mutation_Type.random,
+                 mutation_percent: float = 25.0,
+                 max_mutation: float = 0.05,
+                 stopping_criteria: Stopping_Criteria = Stopping_Criteria(Stopping_Criteria.saturate, 15)):
+        self.chromosome_size: int = chromosome_size
+        self.parent_selection_type: OptimizationSettings.Parent_Selection_Type = parent_selection_type
+        self.generation_limit: int = generation_limit
+        self.num_parents: int = num_parents
+        self.k_tournament: int = k_tournament
+        self.crossover_type: OptimizationSettings.Crossover_Type = crossover_type
+        self.elitism: int = elitism
+        self.mutation_type: OptimizationSettings.Mutation_Type = mutation_type
+        self.mutation_percent: float = mutation_percent
+        self.max_mutation: float = max_mutation
+        self.stopping_criteria: OptimizationSettings.Stopping_Criteria = stopping_criteria
+
+        self._fitness: float = 0
+
+    def as_list(self) -> list[str]:
+        """
+
+        Returns all optimization settings and the stored, associated fitness value as a list that
+        can be outputted as a row to a CSV spreadsheet.
+
+        :return: a list containing each hyperparameter as a string
+
+        """
+
+        out_list: list[str] = [str(self.chromosome_size), str(self.parent_selection_type), str(self.generation_limit),
+                               str(self.num_parents), str(self.k_tournament), str(self.crossover_type),
+                               str(self.elitism), str(self.mutation_type), str(self.mutation_percent),
+                               str(self.max_mutation), str(self.stopping_criteria), str(self._fitness)]
+        return out_list
+
+    def set_fitness(self, fitness: float) -> None:
+        """
+
+        Set the fitness value of this hyperparameter configuration.
+
+        :param float fitness: the fitness achieved by this configuration
+
+        """
+
+        self._fitness = fitness
+
+
+class GeneticOptimization(BaseOptimization):
+    """
+
+    GeneticOptimization uses the PyGAD module to implement a genetic algorithm-based optimization sequence.
+
+    Briefly, GA begins by evaluating an initial population, selecting certain potential solutions (chromosomes) to be
+    parents, creating offspring solutions from the parents, and repeating for a certain number of generations or
+    until a stopping condition is met.
+
+    To learn how GA works, read the resources enumerated at the top of this file (`genetic.py`).
+
+    To modify GA's default hyperparameters, modify the default parameters of OptimizationSettings' constructor.
+
+    To evaluate different hyperparameter configurations, use the "run_hyperparameter_search"
+    method in `run_simulation.py`. To view already evaluated hyperparameters, view the "Hyperparameter Search"
```

Comment:
nitpick: it feels natural to have run_hyperparam_search in genetic.py with the rest of genetic op stuff considering it isnt being used in run_sim.py